GHE_main.cpp

```diff
@@ -0,0 +1,237 @@
+        // need to ensure variables passed to g are cosistent with what is required
+        total = total + (qi - qi1)* g_data[i];
+        ++i;}
+    return total;
+}
+
+/*---------------
+Main function
+----------------*/
+output_data simple_GHE(double Tin, double mdot)
+{
+//    //Debugging print functions:
+//    int j = 0;
+//    while (j < 5){
+//        printf("q at import for index %d is %f\n",j,q_array[j]);
+//        j = j + 1;
+//    }
+//    j = 0;
+//    while (j < 5){
+//        printf("g at import for index %d is %f\n",j,g_data[j]);
+//        j = j + 1;
+//    }
+    
+    // loading JSON data. In development
+    json_data input_data = read_json(); //this will be a function that reads and returns the data from the JSON file. See sndbx for variable type
+    vector<double> q_data = get<2>(input_data.indexed_data);
+    double cp = input_data.cp;
+    double H = input_data.H;
+    double Rb = input_data.Rb;
+    double ks = input_data.ks;
+    double Ts = input_data.Ts;
+    
+    //loading g_data
+    tuple<vector<double>, vector<int>::size_type> g_data = g_expander(input_data.indexed_data, ts);
+    
+    //Defining variables for use in fucntion
+    int n = get<1>(g_data);
+    qn1 = q_data[n];
+    n = n+1;
+    ts = pow(H,2)/(9*alphas);
+    alphas = 1*pow(10,-6);
+    
+
+    
+    //Debugging print functions:
+    printf("n is %d\n",n);
+    printf("qn1 is %f\n",qn1);
+
+    //eqn 1.2
+    // need to ensure variables passed to g are cosistent with what is required
+    gn = get<0>(g_data)[n];
+    
+    // eqn 1.3
+    c0 = 1/(2 * M_PI * ks);
+    
+    //eqn 1.11
+    c1 = summation(n, q_data, get<0>(g_data));
+
+    /* Main functions to solve: 1.13, 1.12, 1.14 (In that order)*/
+    // 1.13
+    qn = ( (Tin - Ts) + ( ((qn1*gn)/c0) - (c1/c0) ) ) / ( (0.5 * (H/(mdot * cp))) + (gn / c0) + Rb);
+    // 1.12
+    Tf = Ts + c0*(((qn - qn1) * gn) + c1) + qn * Rb;
+    //1.14
+    Tout = Tf - 0.5*( (qn * H)/(mdot * cp) );
+    
+    printf("qn before export is %f\n",qn);
+    printf("Tf before export is %f\n",Tf);
+    printf("Tout before export is %f\n",Tout);
+    
+    //updating q data and writing to structure for export from function:
+    q_data[n] = qn;
+    output_data outputs;
+    outputs.temp_out = Tout;
+    outputs.q_out = q_data;
+    
+//    Debugging print functions:
+//    j = 0;
+//    while (j < 6){
+//        printf("q before export for index %d is %f\n",j,q_array[j]);
+//        j = j + 1;
+//    }
+//    j = 0;
+//    while (j < 5){
+//        printf("g before export for index %d is %f\n",j,g_data[j]);
+//        j = j + 1;
+//    }
+    return outputs ;
+}
+
+
+int main()
+{
+    
+    double Tin = 20;
+    double mdot = 0.0002;
+    // required to determine the index of the q data
+    output_data outputs = simple_GHE(Tin, mdot);
+   
+    //reading output data
+    int j = 0;
+    while (j < 6){
+        printf("q after export for index %d is %f\n",j,outputs.q_out[j]);
+        j = j + 1;
+    }
```

Comment:
If we were targeting C++20, this `int` and `while` loop could be really succinctly written as:

```
for (int j = 0; auto const & q : outputs.q_out) {
  std::cout << std::format("q after export for index {} is {}\n", j, q);
}
```

But alas, we are not......for now :upside_down_face: